src/main/java/command/DoneCommand.java

```diff
@@ -0,0 +1,54 @@
+package command;
+
+import exception.DoneException;
+import storage.Storage;
+import task.Task;
+import task.TaskList;
+import ui.Ui;
+
+/**
+ * Command for specifying that a task is done.
+ */
+public class DoneCommand extends Command {
+
+    protected String[] input;
+
+    public DoneCommand(String[] input) {
+        this.input = input;
+    }
+
+    /**
+     * Execute 'ticking' a task in user's tasks list.
+     * Output what is needed.
+     * @param tasks the TaskList.
+     * @param ui the User Interface which responsible for every output printing.
+     * @param storage user's hard disk storage.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DoneException {

```

Comment:
It seems inefficient to parse the input in the execute method for the Command class. Wouldn't it be cleaner and more efficient to parse it in entirely in the Parser first class and then have a class variable of type 'Task' in the DoneCommand? (rather than having a class variable of type String[] - which you still have to parse!)